app/battle.py

```diff
@@ -0,0 +1,14 @@
+from app.knights import Knight
+
+
+def battle(knight1: Knight, knight2: Knight) -> None:
+
+    knight2.hp -= max(0, knight1.power - knight2.protection)
+
+    if knight2.hp <= 0:
+        knight2.hp = 0

```

Comment:
```suggestion
    knight2.hp = max(knight2.hp, 0)
```
In this case use also max
the same for knight1